database/ydb/ydb.go

```diff
@@ -0,0 +1,311 @@
+	})
+	if err != nil {
+		return nil, err
+	}
+	return px, nil
+}
+
+func (db *YDB) parseCredentialsOptions(url *url.URL, query url.Values) (credentials ydb.Option) {
+	switch {
+	case query.Has(queryParamAuthToken):
+		credentials = ydb.WithAccessTokenCredentials(query.Get(queryParamAuthToken))
+	case url.User != nil:
+		user := url.User.Username()
+		password, _ := url.User.Password()
+		credentials = ydb.WithStaticCredentials(user, password)
+	default:
+		credentials = ydb.WithAnonymousCredentials()
+	}
+	url.User = nil
+	return credentials
+}
+
+func (db *YDB) parseTLSOptions(_ *url.URL, query url.Values) (options []ydb.Option, err error) {
+	if query.Has(queryParamTLSCertificateAuthorities) {
+		options = append(options, ydb.WithCertificatesFromFile(query.Get(queryParamTLSCertificateAuthorities)))
+	}
+	if query.Has(queryParamTLSInsecureSkipVerify) {
+		options = append(options, ydb.WithTLSSInsecureSkipVerify())
+	}
+	if query.Has(queryParamTLSMinVersion) {
+		switch query.Get(queryParamTLSMinVersion) {
+		case "1.0":
+			options = append(options, ydb.WithMinTLSVersion(tls.VersionTLS10))
+		case "1.1":
+			options = append(options, ydb.WithMinTLSVersion(tls.VersionTLS11))
+		case "1.2":
+			options = append(options, ydb.WithMinTLSVersion(tls.VersionTLS12))
+		case "1.3":
+			options = append(options, ydb.WithMinTLSVersion(tls.VersionTLS13))
+		default:
+			return nil, ErrUnsupportedTLSVersion
+		}
+	}
+	return options, nil
+}
+
+func (db *YDB) Close() error {
+	return db.driver.Close(context.TODO())
+}
+
+func (db *YDB) Run(migration io.Reader) error {
+	rawMigrations, err := io.ReadAll(migration)
+	if err != nil {
+		return err
+	}
+
+	res, err := db.driver.Scripting().Execute(context.TODO(), string(rawMigrations), nil)
+	if err != nil {
+		return err
+	}
+	return res.Close()
+}
+
+func (db *YDB) SetVersion(version int, dirty bool) error {
+	deleteVersionQuery := fmt.Sprintf(`
+		DELETE FROM %s 
+	`, db.config.MigrationsTable)
+
+	insertVersionQuery := fmt.Sprintf(`
+		INSERT INTO %s (version, dirty, created) VALUES (%d, %t, CurrentUtcTimestamp())
+	`, db.config.MigrationsTable, version, dirty)
+
+	ctx := context.TODO()
+	err := db.driver.Query().DoTx(ctx, func(ctx context.Context, tx query.TxActor) error {
+		if err := tx.Exec(ctx, deleteVersionQuery); err != nil {
+			return err
+		}
+		// Also re-write the schema version for nil dirty versions to prevent
+		// empty schema version for failed down migration on the first migration
+		// See: https://github.com/golang-migrate/migrate/issues/330
+		if version >= 0 || (version == database.NilVersion && dirty) {
+			if err := tx.Exec(ctx, insertVersionQuery); err != nil {
+				return err
+			}
+		}
+		return nil
+	}, query.WithTxSettings(query.TxSettings(query.WithSerializableReadWrite())))
+	return err
+}
+
+func (db *YDB) Version() (version int, dirty bool, err error) {
+	ctx := context.TODO()
+
+	getVersionQuery := fmt.Sprintf(`
+		SELECT version, dirty FROM %s ORDER BY version DESC LIMIT 1
```

Comment:
You clear the table in `SetVersion()`, then maybe we do not need `ORDER BY DESC LIMIT 1` here?

Reply:
`ORDER BY DESC` is not needed, and actually `LIMIT 1` also not needed
but all other adapters use `LIMIT 1`, so I left this for unification